src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs

```diff
@@ -857,265 +890,125 @@ public void RenderDrawables(VisualizationEventArgs e)
             renderTimer.Start();
 #endif
 
-            SendGraphicsToView(points, lines, linesSelected, mesh, text);
+            SendGraphicsToView(points, lines, linesSel, mesh, meshSel, perVertexMesh, text);
 
             //DrawTestMesh();
         }
 
-        private static LineGeometry3D InitLineGeometry()
-        {
-            var lines = new LineGeometry3D
-            {
-                Positions = new Vector3Collection(),
-                Indices = new IntCollection(),
-                Colors = new Color4Collection()
-            };
-
-            return lines;
-        }
-
-        private static PointGeometry3D InitPointGeometry()
-        {
-            var points = new PointGeometry3D()
-            {
-                Positions = new Vector3Collection(),
-                Indices = new IntCollection(),
-                Colors = new Color4Collection()
-            };
-
-            return points;
-        }
-
-        private static MeshGeometry3D InitMeshGeometry()
+        private void AggregateRenderPackages(IEnumerable<HelixRenderPackage> packages, 
+            PointGeometry3D points, LineGeometry3D lines, 
+            LineGeometry3D linesSel, MeshGeometry3D mesh, MeshGeometry3D perVertexMesh, MeshGeometry3D meshSelected, BillboardText3D text)
```

Comment:
Let's turn this into a single class, like `RenderParams`, because this method signature is impossible to read and will cause developers to make mistakes.  Maintenance should trump optimization here.  Plus, the overhead of creating the class is nothing.